tests/unit_tests/test_public_api.py

```diff
@@ -212,6 +212,33 @@ def test_artifact_download_logger():
             termlog.assert_not_called()
 
 
+def test_update_aliases_on_artifact(user, relay_server, wandb_init):
+    project = "test"
+    run = wandb_init(entity=user, project=project)
+    artifact = wandb.Artifact("test-artifact", "test-type")
+    with open("boom.txt", "w") as f:
+        f.write("testing")
+    artifact.add_file("boom.txt", "test-name")
+    run.log_artifact(artifact, aliases=["best"])
+    artifact.wait()
+    run.finish()
+
+    artifact = Api().artifact(
+        name=f"{user}/{project}/test-artifact:v0", type="test-type"
+    )
+    artifact.aliases = ["staging"]
+    artifact.aliases.append("boom")
+    artifact.save()
+
+    artifact = Api().artifact(
```

Comment:
nit: seems like a big part of this test is when the artifact is linked to another portfolio that the mutations to the alias list only apply to the artifact collection for which the artifact was fetched.

Reply:
maybe worth extending the test

Reply:
done